core/src/test/java/org/elasticsearch/search/sort/FieldSortBuilderTest.java

```diff
@@ -0,0 +1,184 @@
+        QueryBuilder nested = null;
+        while (nested == null || nested.equals(builder.getNestedFilter())) {
+            switch (randomInt(2)) {
+            case 0:
+                nested = new MatchAllQueryBuilder();
+                break;
+            case 1:
+                nested = new IdsQueryBuilder();
+                break;
+            default:
+            case 2:
+                nested = new TermQueryBuilder(randomAsciiOfLengthBetween(1, 10), randomAsciiOfLengthBetween(1, 10));
+                break;
+            }
+            nested.boost((float) randomDoubleBetween(0, 10, false));
+        }
+        builder.setNestedFilter(nested);
+    }
+
+    private void nestedPath(FieldSortBuilder builder) {
+        String nestedPath = randomAsciiOfLengthBetween(1, 10);
+        while (nestedPath.equals(builder.getNestedPath())) {
+            nestedPath = randomAsciiOfLengthBetween(1, 10);
+        }
+        builder.setNestedPath(nestedPath);
+    }
+
+    private void modes(FieldSortBuilder builder) {
+        String[] modes = {"min", "max", "avg", "sum"};
+        String mode = randomFrom(modes);
+        while (mode.equals(builder.sortMode())) {
+            mode = randomFrom(modes);
+        }
+        builder.sortMode(mode);
+    }
+    
+    private void missing(FieldSortBuilder builder) {
+        Object missing = null;
+        Object otherMissing = null;
+        if (builder.missing() instanceof BytesRef) {
+            otherMissing = ((BytesRef) builder.missing()).utf8ToString();
+        } else {
+            otherMissing = builder.missing();
+        }
+
+        while (missing == null || missing.equals(otherMissing)) {
+          int missingId = randomIntBetween(0, 3);
+          switch (missingId) {
+          case 0:
+              missing = ("_last");
+              break;
+          case 1:
+              missing = ("_first");
+              break;
+          case 2:
+              missing = randomAsciiOfLength(10);
+              break;
+          case 3:
+              missing = randomInt();
+              break;
+          default:
+              throw new IllegalStateException("Unknown missing type.");
+              
+          }
+        }
+        builder.missing(missing);
+    }
+    
+    public void unmappedType(FieldSortBuilder builder) {
+        String type = randomAsciiOfLengthBetween(1, 10);
+        while (type.equals(builder.unmappedType())) {
+            type = randomAsciiOfLengthBetween(1, 10);
+        }
+        builder.unmappedType(type);
+    }
+    
+    public void order(FieldSortBuilder builder) {
+        SortOrder order = SortOrder.ASC;
+        if (order.equals(builder.order())) {
+            builder.order(SortOrder.DESC);
+        } else {
+            builder.order(SortOrder.ASC);
+        }
+    }
+
+    @Override
+    protected FieldSortBuilder mutate(FieldSortBuilder original) throws IOException {
+        FieldSortBuilder mutated = new FieldSortBuilder(original);
```

Comment:
The advantage of doing this via serialization instead of copy constructor would be that this is a deep copy then, this way we only copy the references.